### ISSUE TITLE
Adding dotnet build /t:help  support for individual effects

### ISSUE THREAD

**Issue by Anipik** [dotnet/runtime#38281]
The purpose here to provide list of properties that developers can use to build different scenarios,

eg TargetOS, TargetArchitecture, Tfm, BuildTargetFramework, etc.

cc @safern @ViktorHofer @ericstj

**Comment by ViktorHofer** [comment]
How does a developer know that the `help` target exists? If the developer already has to read the docs to find out about it, why wouldn't we prefer that place over a help target to document these properties?

**Comment by safern** [comment]
We thought that adding `/t:help` might be really helpful for people to remember what is available for that specific project. Based on project type we could expand available common targets, how to run tests, what properties are available, etc.

**Comment by ViktorHofer** [comment]
So again, how would a developer know that the help target exists without finding it in the docs? I don't see much value in doing that and would instead want to focus on improving the existing docs. Some of those are long and hard to read.

**Comment by danmosemsft** [comment]
I don't know whether I'm a representative user, but I am confident I can remember `/t:help`, but will certainly forget the various names of the properties and their possible values (viz: the recent discussion about how to build for FreeBSD). And they may change in future.

The docs are important, but for simple mainstream commands I'd like to get the job done quickly without heading to look at docs., This is exactly what we did for build.cmd/sh and it was so useful.

**Comment by ViktorHofer** [comment]
So I probably missed the point. I though this is about improving the first time contributor scenario.

**Comment by danmosemsft** [comment]
The first time contributor is inevitably going to have to walk through the docs. The idea behind something like this is to try to reduce the number of times they have to return to those docs, I guess.

**Comment by safern** [comment]
Yeah the point of this is once you know how the repo build works, you already read the docs, so that you don’t have to go back to the docs and just remember about /t:help and that will help you without having to leave the terminal at all.

A lot of people reach out why doesn’t /p:TargetArch work, because they forget it is /p:TargetArchitecture. Also they forget various combinations to cross compile, how to make a project that has iOS specific configs use that OS, etc.
